rust/arrow/src/datatypes.rs

```diff
@@ -930,6 +932,7 @@ impl DataType {
                 json!({"name": "fixedsizebinary", "byteWidth": byte_width})
             }
             DataType::Struct(_) => json!({"name": "struct"}),
+            DataType::Union(_) => json!({"name": "union"}),
```

Comment:
Some matches for `Union(_)` are missing, but I'm happy if we address them when dealing with IPC, as they're more relevant then.